flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNITCase.java

```diff
@@ -93,8 +114,8 @@ private void deployPerjob(YarnConfigOptions.UserJarInclusion userJarInclusion) t
 			true)) {
 
 			yarnClusterDescriptor.setLocalJarPath(new Path(flinkUberjar.getAbsolutePath()));
-			yarnClusterDescriptor.addShipFiles(Arrays.asList(flinkLibFolder.listFiles()));
-			yarnClusterDescriptor.addShipFiles(Arrays.asList(flinkShadedHadoopDir.listFiles()));
+			yarnClusterDescriptor.addShipFiles(Collections.singletonList(flinkLibFolder));
```

Comment:
I don't stick to revert this but want to know that, we don't change anything by this diff right? IIRC Flink will iterate over a directory.

Reply:
Yes, you are right. Flink will iterate over a directory. So i just make the test to keep the same behavior with `YarnClusterDescriptor`. `YarnClusterDescriptor` add the `lib` folder as ship file, not all the files.

We find the corresponding pre-uploaded file by the same relative path with local file. So it is important to test the real behavior.